client/ayon_harmony/plugins/create/create_workfile.py

```diff
@@ -1,100 +1,27 @@
-        host_name = self.create_context.host_name
-
-        if current_instance is None:
-            product_name = self.get_product_name(
-                project_name,
-                folder_entity,
-                task_entity,
-                variant,
-                host_name,
-            )
-            data = {
-                "folderPath": folder_entity["path"],
-                "task": task_entity["name"],
-                "variant": variant
-            }
-            data.update(
-                self.get_dynamic_data(
-                    project_name,
-                    folder_entity,
-                    task_entity,
-                    variant,
-                    host_name,
-                    current_instance)
-            )
-            self.log.info("Auto-creating workfile instance...")
-            current_instance = CreatedInstance(
-                self.product_type, product_name, data, self
-            )
-            self._add_instance_to_context(current_instance)
-        elif (
-            current_instance["folderPath"] != folder_entity["path"]
-            or current_instance["task"] != task_entity["name"]
-        ):
-            # Update instance context if is not the same
-            product_name = self.get_product_name(
-                project_name,
-                folder_entity,
-                task_entity,
-                variant,
-                host_name,
-            )
-
-            current_instance["folderPath"] = folder_entity["path"]
-            current_instance["task"] = task_entity["name"]
-            current_instance["productName"] = product_name
-
-        current_instance.transient_data["node"] = self._node_name
-
-    def collect_instances(self):
-        cache = self.cache_instance_data(self.collection_shared_data)
-        for node in cache.get("harmony_cached_instance_data").get(
-                self.identifier, []):
-            data = cache.get("harmony_cached_scene_data")[node]
-            created_instance = CreatedInstance.from_existing(data, self)
-            created_instance.transient_data["node"] = self._node_name
-            self._add_instance_to_context(created_instance)
-
-    def update_instances(self, update_list):
-        for created_inst, _changes in update_list:
-            lib.imprint(self._node_name, created_inst.data_to_store())
-
-    def remove_instances(self, instances):
-        for instance in instances:
-            scene_data = lib.get_scene_data()
-            scene_data.pop(self._node_name, None)
-            lib.set_scene_data(scene_data)
-
-            self._remove_instance_from_context(instance)
+    def apply_settings(self, project_settings):
+        plugin_settings = (
+            project_settings["harmony"]["create"]["CreateWorkfile"]
+        )
+        self.enabled = plugin_settings["enabled"]
+        self.default_variants = plugin_settings["default_variants"]
+        self.active_on_create = plugin_settings["active_on_create"]
```

Comment:
This would be redundant if you'd set `settings_category = "harmony"`, right?

Reply:
Yeah, it works, wasn't sure it if will. Removed, pushed to `plugin`.